### PR TITLE
Simplify frontend environment config

### PR DESCRIPTION
**Pull request by SamChou19815**
### Summary 

This PR leverages the config from create-react-app to make frontend dev experience better.

### Test Plan 

Check preview site to ensure that the site can still load and login.

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
     "test": "react-scripts test",
     "eject": "react-scripts eject"
   },
+  "proxy": "http://localhost:9000",
```
Comment (SamChou19815) [inline frontend/package.json:34]: See https://create-react-app.dev/docs/proxying-api-requests-in-development/

File: frontend/src/environment.ts [hunk 2]
```diff
@@ -1,15 +1,9 @@
...
-    : 'http://localhost:9000/.netlify/functions/api/'
-};
+/** Switch to true when using prod API for dev. Remember to change it back before commit. */
+const useProdBackendForDev = false;
 
-export default environment;
+export const backendURL =
+  isProduction || !useProdBackendForDev
+    ? '/.netlify/functions/api'
+    : 'http://idol.cornelldti.org/.netlify/functions/api';
```
Comment (SamChou19815) [inline frontend/src/environment.ts:9]: Somehow this is still not working :(

Comment (henry-li-06) [reply frontend/src/environment.ts:9]: There was an issue w/ the production URLs that should've been fixed since, it should work now

**Review by henry-li-06 — APPROVED** [review]
LGTM

**Pull request merged**